lib/matplotlib/widgets.py

```diff
@@ -2003,117 +2051,254 @@ def new_axes(self, ax):
+        Positions of handles in data coordinates.
+    direction : {"horizontal", "vertical"}
+        Direction of handles, either 'vertical' or 'horizontal'
+    line_props : dict
+        Additional line properties. See `matplotlib.lines.Line2D`.
+    """
 
-        minv, maxv = v, self.pressv
-        if minv > maxv:
-            minv, maxv = maxv, minv
+    def __init__(self, ax, positions, direction, line_props=None,
+                 useblit=True):
+        self.ax = ax
+
+        _api.check_in_list(['horizontal', 'vertical'], direction=direction)
+        self._direction = direction
+
+        if line_props is None:
+            line_props = {}
+        line_props.update({'visible': False, 'animated': useblit})
+
+        line_fun = ax.axvline if self.direction == 'horizontal' else ax.axhline
+
+        self.artists = [line_fun(p, **line_props) for p in positions]
+
+    @property
+    def positions(self):
+        """Positions of the handle in data coordinates."""
+        method = 'get_xdata' if self.direction == 'horizontal' else 'get_ydata'
+        return [getattr(line, method)()[0] for line in self.artists]
+
+    @property
+    def direction(self):
+        """Direction of the handle: 'vertical' or 'horizontal'."""
+        return self._direction
+
+    def set_data(self, positions):
+        """
+        Set x positions of handles
+
+        Parameters
+        ----------
+        positions : tuple of length 2
+            Set the positions of the handle in data coordinates
+        """
+        method = 'set_xdata' if self.direction == 'horizontal' else 'set_ydata'
+        for line, p in zip(self.artists, positions):
+            getattr(line, method)([p, p])
+
+    def set_visible(self, value):
+        """Set the visibility state of the handles artist."""
+        for artist in self.artists:
+            artist.set_visible(value)
+
+    def set_animated(self, value):
+        """Set the animated state of the handles artist."""
+        for artist in self.artists:
+            artist.set_animated(value)
+
+    def closest(self, x, y):
+        """
+        Return index and pixel distance to closest handle.
+
+        Parameters
+        ----------
+        x, y : float
+            x, y position from which the distance will be calculated to
+            determinate the closest handle
+
+        Returns
+        -------
+        index, distance : index of the handle and its distance from
+            position x, y
+        """
         if self.direction == 'horizontal':
-            self.rect.set_x(minv)
-            self.rect.set_width(maxv - minv)
+            p_pts = np.array([
+                self.ax.transData.transform((p, 0))[0] for p in self.positions
+                ])
+            dist = abs(p_pts - x)
         else:
-            self.rect.set_y(minv)
-            self.rect.set_height(maxv - minv)
+            p_pts = np.array([
+                self.ax.transData.transform((0, p))[1] for p in self.positions
```

Comment:
How can you be sure that 0 in the other coordinate doesn't produce some invalid transform (e.g., if the scale is log)?

Reply:
I don't know but I couldn't find an issue. Do you have an example?

On the topic of log scale, it seems that `drag_from_anywhere` doesn't behave as we would expect and the same issue occurs with `RectangleSelector`/`EllipseSelector`.

Reply:
I tried it out in the example, and it seems to be fine for log scale in the other coordinate. It must just produce some silent NaNs.

Log scale on the span selector coordinate works "as expected" in that it's moved in the data space, and gets bigger/smaller based on the position in log space. I don't know if we want to fix it or not (or if it's really a fix exactly.)

Reply:
It feels to me as if we should move in screen space instead?

Reply:
I don't think any widgets do that (e.g., see discussion on #19864).

Reply:
OK, I guess we can keep that discussion for another time then (and move them all at once).

Reply:
Yes, and it would need to be done for all selectors.